bin/dry-run.rb

```diff
@@ -447,15 +454,19 @@ def handle_dependabot_error(error:, dependency:)
 always_clone = Dependabot::Utils.
                always_clone_for_package_manager?($package_manager)
 if $options[:clone] || always_clone
-  $repo_contents_path = Dir.mktmpdir
-  puts "=> cloning into #{$repo_contents_path}"
+  $repo_contents_path = File.expand_path(File.join("tmp", $repo_name.split("/")))
 end
 
 fetcher = Dependabot::FileFetchers.for_package_manager($package_manager).
           new(source: $source, credentials: $options[:credentials],
               repo_contents_path: $repo_contents_path)
 $files = if $options[:clone] || always_clone
-           fetcher.clone_repo_contents
+           unless $options[:cache_steps].include?("files") && Dir.exist?(File.join($repo_contents_path, ".git"))
```

Comment:
Nit: it's a weird smell that this duplicates the check that `clone_repo_contents` does internally https://github.com/dependabot/dependabot-core/blob/e7c1929d7f850f8516311cc41753958cb6b980d2/common/lib/dependabot/file_fetchers/base.rb#L476

Maybe `clone_repo_contents` should return a status?

Reply:
Oh good spot, forgot we did that. Will have a think about improving the cloning behaviour, we currently always ignore the `--commit` option for example.